spring-cloud-skipper-server/src/main/java/org/springframework/cloud/skipper/server/service/ReleaseService.java

```diff
@@ -167,7 +171,10 @@ private void validateInstallRequest(InstallRequest installRequest) {
 
 	protected Release install(PackageMetadata packageMetadata, InstallProperties installProperties) {
 		Assert.notNull(packageMetadata, "Can't download package, PackageMetadata is a null value.");
-		Release release = createInitialRelease(installProperties, this.packageService.downloadPackage(packageMetadata));
+		Release existingDeletedRelease = this.releaseRepository.findLatestReleaseIfDeleted(installProperties.getReleaseName());
```

Comment:
I don't think we want this behavior for install.  The first use-case, is to be able to rollback a release that is deleted.  That seems analogous to the 'trash can' concept in an OS, we can restore something we put in the trash.

In helm, there is a specific option wrt to supporting this on install, the option `--replace`

From the helm description of that install option, `re-use the given name, even if that name is already used. This is unsafe in production`

I do wonder the motivation around that last comment, not safe in production.  I believe doing a `rollback` followed by an `upgrade` would be the logical equivalent.

Created https://github.com/spring-cloud/spring-cloud-skipper/issues/207 to track

Reply:
I will add some info logging for now to indicate which code pathway is being taken.